src/main/java/uk/gov/justice/digital/delius/service/RiskService.java

```diff
@@ -1,12 +1,25 @@
 package uk.gov.justice.digital.delius.service;
 
+import org.springframework.data.domain.PageRequest;
 import org.springframework.stereotype.Service;
+import uk.gov.justice.digital.delius.controller.NotFoundException;
 import uk.gov.justice.digital.delius.data.api.MappaDetails;
+import uk.gov.justice.digital.delius.jpa.standard.repository.RegistrationRepository;
+import uk.gov.justice.digital.delius.transformers.MappaDetailsTransformer;
 
-// TODO DT-1542 Implement this service
 @Service
 public class RiskService {
+    private final RegistrationRepository registrationRepository;
+
+    public RiskService(RegistrationRepository registrationRepository) {
+        this.registrationRepository = registrationRepository;
+    }
+
     public MappaDetails getMappaDetails(Long offenderId) {
-        return null;
+        final var reg = registrationRepository.findActiveMappaRegistrationByOffenderId(offenderId, PageRequest.of(0, 1));
+            return reg.stream()
+                .findFirst()
+                .map(registration -> MappaDetailsTransformer.mappaDetailsOf(registration))
+                .orElseThrow(() -> new NotFoundException("MAPPA details for offender not found"));
     }
```

Comment:
I can't see an alternative to a 404 but I wonder in longer term if we need a clear way of saying what entity is not found, something like

`"reason": "offender"`
or 
`"reason": "mappa"`

Maybe we need some feedback from clients since I have always found this possibly a cause for confusion.
Having said that I suspect the fact a client is making this call they have already established the offender exists so maybe it doesn't matter.

Either nothing to do for this change

Reply:
I know what you mean, but I think it's a rare occasion that the client cares why - I haven't come across it much.  So yes I think we only do that if the client asks for it.